app/assets/javascripts/discourse/app/mixins/composer-upload-uppy.js

```diff
@@ -319,11 +323,13 @@ export default Mixin.create(ExtendableUploader, UppyS3Multipart, {
       // only do the manual cancelling work if the user clicked cancel
       if (this.userCancelled) {
         Object.values(this.placeholders).forEach((data) => {
-          this.appEvents.trigger(
-            `${this.eventPrefix}:replace-text`,
-            data.uploadPlaceholder,
-            ""
-          );
+          run(() => {
```

Comment:
Shouldn't all `this._uppyInstance.on` callbacks be wrapped in `run()`?

Reply:
I think so, but I didn't want to change too much right now.